### ISSUE TITLE
Automatically add -c command option for OpenApiGenerator

### ISSUE THREAD

**Issue by johnwc** [christianhelle/apiclientcodegen#610]
Can you add some logic so that if within the same directory as the api spec file, exists a config file named `<api spec filename>.config.yaml`, you then add the `-c <api spec filename>.config.yaml` command line option when running openapi-generator-cli.jar? Having the ability to opt out of this functionality via Tools->Options menu.

That way we can add all of our config to a file that can be store in source control. 

### Example:
```
Directory: c:\project\
ApiClient.yaml
ApiClient.config.yaml
```

**Comment by christianhelle** [comment]
@johnwc thanks for suggesting this. It sounds simple and easy enough to implement. I'm short on time these days so it won't be any time soon, but I'll definitely look into this

**Comment by christianhelle** [comment]
@johnwc I'm going to start working on this. Do you have an example config file I can test with? This way, we're sure that your scenario will work

**Comment by johnwc** [comment]
The config file would be the config file the API generator would expect to read in via the -c command line option. So you can make up one that fits your needs.

**Comment by christianhelle** [comment]
@johnwc Do we just make up a file format based on available config options described in https://openapi-generator.tech/docs/generators/csharp#config-options ?

**Comment by christianhelle** [comment]
Ok, I think I fully understand now. It's described under [Configuration File](https://openapi-generator.tech/docs/usage/#configuration-file) in the OpenAPI Generator usage docs

**Comment by christianhelle** [comment]
Is this what you had in mind @johnwc 

![image](https://github.com/christianhelle/apiclientcodegen/assets/710400/9a3c5d1c-91fb-4f63-8aad-f63c47429045)

**Comment by christianhelle** [comment]
And of course it works with YAML too

![image](https://github.com/christianhelle/apiclientcodegen/assets/710400/8b59124b-8d95-420c-8b12-c9719be3f292)

**Comment by johnwc** [comment]
Yes, perfect! Though as long as it supports using json config with yaml API schema.

**Comment by christianhelle** [comment]
@johnwc as it is right now, the config file needs to be in the same file format as the OpenAPI spec file

Like
OpenAPI.json
OpenAPI.config.json

Or
OpenAPI.yaml
OpenAPI.config.yaml

It not hard to add support for mix matching file formats and I can also implement that later tonight, if needed.

**Comment by johnwc** [comment]
We wouldn't be able to use it unless it supported using json config with yaml spec file.

**Comment by christianhelle** [comment]
> We wouldn't be able to use it unless it supported using json config with yaml spec file.

Like this?

![image](https://github.com/christianhelle/apiclientcodegen/assets/710400/d72b1fc5-29d8-4635-a972-3c8ce34b025a)

**Comment by johnwc** [comment]
Yes, that looks great

**Comment by christianhelle** [comment]
@johnwc This feature is now released to the marketplace in v1.8.4

**Comment by christianhelle** [comment]
@johnwc I hope it's not too much to ask but I would be very grateful for a rating or review on the [Visual Studio Marketplace](https://bit.ly/rapicgen-vs2022)
